lib/releases.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) OpenTofu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+class Build {
+  constructor (name, url) {
+    this.name = name;
+    this.url = url;
+  }
+}
+
+class Release {
+  constructor (releaseMeta) {
+    this.version = releaseMeta.tag_name.replace('v', '');
+    this.builds = releaseMeta.assets.map(asset => new Build(asset.name, asset.browser_download_url));
+  }
+
+  getBuild (platform, arch) {
+    const requiredName = `tofu_${this.version}_${platform}_${arch}.zip`;
+    return this.builds.find(build => build.name === requiredName);
+  }
+}
+
+/**
+ * Fetches the top 30 releases sorted in desc order.
+ *
+ */
+async function fetchReleases () {
+  const url = 'https://api.github.com/repos/opentofu/opentofu/releases';
+
+  const resp = await fetch(url, {
+    headers: {
+      Accept: 'application/vnd.github+json',
+      'X-GitHub-Api-Version': '2022-11-28'
+    }
+  });
+
+  if (!resp.ok) {
+    throw Error('failed fetching releases');
+  }
```

Comment:
I'd rather leave it as is to avoid side effects because. If github API responds something other than 200, not much can be done. However, feel free to raise an issue with the proposal how would improve error handling. Although, it appears to be an overthinking at this point.